rpcserver.go

```diff
@@ -7034,6 +7034,92 @@ func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
 	case len(firstHopRecords) > 0:
 		// Continue below.
 
+	case req.RfqId != nil:
+		// Check if the provided rfq ID matches the expected length.
+		if len(req.RfqId) != 32 {
+			return fmt.Errorf("rfq id length must be 32")
+		}
+
+		// Now let's try to perform an internal lookup to see if there's
+		// an actual quote on this ID.
+		var rfqID rfqmsg.ID
+		copy(rfqID[:], req.RfqId)
+
+		var quote rfqmsg.SellAccept
+		for id, q := range r.cfg.RfqManager.PeerAcceptedSellQuotes() {
+			if id == rfqID.Scid() {
+				quote = q
+				break
+			}
+		}
+
+		// This quote ID's scid did not match anything.
+		if quote.ID == [32]byte{} {
+			return fmt.Errorf("scid did not match an accepted " +
+				"quote")
+		}
+
+		// This quote ID's scid matched a different quote.
+		if !bytes.Equal(quote.ID[:], req.RfqId) {
```

Comment:
This would not be needed if we did the comparison on the RFQ ID in the first place.